cumulus/test/service/src/lib.rs

```diff
@@ -860,11 +859,12 @@ pub fn node_config(
 
 	network_config.allow_non_globals_in_dht = true;
 
-	network_config
-		.listen_addresses
-		.push(multiaddr::Protocol::Memory(rand::random()).into());
-
-	network_config.transport = TransportConfig::MemoryOnly;
+	let addr: multiaddr::Multiaddr = format!("/ip4/127.0.0.1/tcp/{}", rand::random::<u16>())
+		.parse()
+		.expect("valid address; qed");
```

Comment:
why is this change needed?

Reply:
We were previously using the memory transport of libp2p. Since litep2p doesn't implement the memory-transport, I've switched this to TCP 🙏

Reply:
Have created this as a followup: https://github.com/paritytech/polkadot-sdk/issues/8691 🙏 Will give it a go locally, but I expect some testing refactoring

Reply:
Oki doki, this should be fixed now on this branch! Have switched to the default port which will be known when polling the network backend